tests/unit/test_scheduler.py

```diff
@@ -0,0 +1,39 @@
+from datetime import datetime, timedelta
+
+import pytest
+from mock import MagicMock, patch
+
+from latigo.model_info import Model
+from latigo.scheduler import Scheduler
+from tests.conftest import SCHEDULER_PREDICTION_DELAY, SCHEDULER_PREDICTION_INTERVAL
+
+DATETIME_UTC_NOW = datetime.fromisoformat("2020-04-10T10:00:00.000000+00:00")
+MODELS = [Model(model_name="model", project_name="project", tag_list=[], target_tag_list=[])]
+
+
+@pytest.fixture
+@patch("latigo.task_queue.kafka.Producer", new=MagicMock())
+@patch("latigo.scheduler.Scheduler._prepare_model_info", new=MagicMock())
+@patch("latigo.scheduler.Scheduler._perform_auth_checks", new=MagicMock())
+def scheduler(schedule_config, monkeypatch) -> Scheduler:
+    monkeypatch.setattr("latigo.scheduler.get_datetime_now_in_utc", MagicMock(return_value=DATETIME_UTC_NOW))
+    scheduler = Scheduler(schedule_config)
+    scheduler.models = MODELS
+    return scheduler
```

Comment:
```suggestion
@pytest.fixture
@patch("latigo.task_queue.kafka.Producer")
@patch("latigo.scheduler.Scheduler._prepare_model_info")
@patch("latigo.scheduler.Scheduler._perform_auth_checks")
@patch("latigo.scheduler.get_datetime_now_in_utc", return_value=DATETIME_UTC_NOW)
def scheduler(schedule_config) -> Scheduler:
    scheduler = Scheduler(schedule_config)
    scheduler.models = MODELS
    return scheduler
```
1. There is a mix of `monkeypatch` and `unittest.mock` libraries usage.  I would suggest to use just one.
2. Unless there are dynamically set public properties, it's more safe to let `patch` use auto-generated spec of a patched object.

Reply:
monkeypatch is used cause `get_datetime_now_in_utc` is imported before we can apply `patch`